eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -39,9 +39,9 @@ case class RemoteChanges(proposed: List[UpdateMessage], acked: List[UpdateMessag
   def all: List[UpdateMessage] = proposed ++ signed ++ acked
 }
 case class Changes(ourChanges: LocalChanges, theirChanges: RemoteChanges)
-case class HtlcTxAndSigs(txinfo: HtlcTx, localSig: ByteVector64, remoteSig: ByteVector64)
-case class PublishableTxs(commitTx: CommitTx, htlcTxsAndSigs: List[HtlcTxAndSigs])
-case class LocalCommit(index: Long, spec: CommitmentSpec, publishableTxs: PublishableTxs)
+case class HtlcTxAndRemoteSig(txinfo: HtlcTx, remoteSig: ByteVector64)
+case class CommitTxAndRemoteSig(commitTx: CommitTx, remoteSig: ByteVector64, htlcTxsAndSigs: List[HtlcTxAndRemoteSig])
```

Comment:
It's a bit weird that now the name doesn't reflect the fact that it contains htlc txs as well.
What it actually contains are all the txs that require both a local and a remote sig, with the corresponding remote sigs.
What about `CommitmentTxsAndRemoteSigs` (I'm not 100% convinced, we can probably do better)?

Or a somewhat different design would be to keep naming this `PublishableTxs(commitTx: CommitTx, remoteSig: ByteVector64, htlcTxsAndSigs: List[HtlcTxAndRemoteSig])` and move the methods to create the fully signed commit tx and fully signed htlc txs inside this class? Because this does contain publishable txs, we just need to add our local sig (which we should always be able to do).

Reply:
If we had decided to not keep the unsigned transactions, we could have removed one level and be consistent with the `RemoteCommit` class:

```scala
case class LocalCommit(index: Long, spec: CommitmentSpec, txid: ByteVector32, remoteCommitSig: ByteVector64, remoteHtlcSigs: List[ByteVector64])
case class RemoteCommit(index: Long, spec: CommitmentSpec, txid: ByteVector32, remotePerCommitmentPoint: PublicKey)
```

That would have been pretty neat. We could do:

```scala
case class LocalCommit(index: Long, spec: CommitmentSpec, commitTxAndRemoteSig: CommitTxAndRemoteSig, htlcTxAndRemoteSig: HtlcTxAndRemoteSig)
case class RemoteCommit(index: Long, spec: CommitmentSpec, txid: ByteVector32, remotePerCommitmentPoint: PublicKey)
```

Reply:
> That would have been pretty neat. We could do:
>
> ```scala
> case class LocalCommit(index: Long, spec: CommitmentSpec, commitTxAndRemoteSig: CommitTxAndRemoteSig, > htlcTxAndRemoteSig: HtlcTxAndRemoteSig)
> case class RemoteCommit(index: Long, spec: CommitmentSpec, txid: ByteVector32, remotePerCommitmentPoint: PublicKey)
> ```

That sounds good, yes!

Reply:
Done with 996355480.